nnvm/src/top/nn/convolution.cc

```diff
@@ -73,15 +73,17 @@ inline bool Conv2DInferShape(const nnvm::NodeAttrs& attrs,
   CHECK_EQ(param.channels % param.groups, 0U)
       << "output channels must divide group size";
 
-  TShape wshape({param.channels / param.groups,
+  // Restore depthwise conv2d kernel layout
```

Comment:
I think the old code is strange and incorrect while your code is straightforward and correct. We can delete this comment.